File: app/src/main/java/me/pwcong/jpstart/component/activity/SettingActivity.java
package me.pwcong.jpstart.component.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.CoordinatorLayout;
import android.support.v7.widget.Toolbar;
import android.view.View;

import butterknife.BindView;
import me.pwcong.jpstart.R;
import me.pwcong.jpstart.component.fragment.SettingFragment;
import me.pwcong.jpstart.rxbus.RxBus;
import me.pwcong.jpstart.rxbus.event.EventContainer;
import me.pwcong.jpstart.rxbus.event.SettingEvent;
import rx.Subscription;
import rx.functions.Action1;

/**
 * Created by Pwcong on 2016/10/6.
 */

public class SettingActivity extends BaseActivity {

    @BindView(R.id.toolbar)
    Toolbar mToolbar;
    @BindView(R.id.layout_root)
    CoordinatorLayout mRootLayout;

    boolean registered = false;

    Subscription subscription;

    @Override
    protected int getViewId() {
        return R.layout.activity_setting;
    }

    @Override
    protected void initVariable(@Nullable Bundle savedInstanceState) {

        if(!registered){

            subscription = RxBus.getDefault().toObserverable(EventContainer.class).subscribe(new Action1<EventContainer>() {
                @Override
                public void call(EventContainer eventContainer) {

                    if(eventContainer.getType()==EventContainer.TYPE_SETTING){

                        SettingEvent event = (SettingEvent) eventContainer.getEvent();
                        showSnackBar(mRootLayout,event.getMsg());
                    }


                }
            });

            registered = true;

        }


        initToolbar();
    }

    @Override
    protected void doAction() {

        getFragmentManager().beginTransaction().replace(R.id.content,new SettingFragment()).commit();

    }

    private void initToolbar(){

        mToolbar.setTitle(R.string.setting);
        setSupportActionBar(mToolbar);

        mToolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        subscription.unsubscribe();
    }
}

File: app/src/main/java/com/github/pwcong/jpstart/ui/activity/SettingActivity.kt
package com.github.pwcong.jpstart.ui.activity

import android.os.Bundle
import android.view.View
import androidx.appcompat.widget.Toolbar
import androidx.coordinatorlayout.widget.CoordinatorLayout
import com.github.pwcong.jpstart.R
import com.github.pwcong.jpstart.databinding.ActivitySettingBinding
import com.github.pwcong.jpstart.rxbus.RxBus
import com.github.pwcong.jpstart.rxbus.event.EventContainer
import com.github.pwcong.jpstart.rxbus.event.SettingEvent
import com.github.pwcong.jpstart.ui.fragment.SettingFragment
import rx.Subscription

class SettingActivity : BaseActivity<ActivitySettingBinding>() {

    private lateinit var mRootLayout: CoordinatorLayout

    private lateinit var mToolbar: Toolbar

    private var registered: Boolean = false

    private var subscription: Subscription? = null

    override fun initViewBinding(): ActivitySettingBinding {
        return ActivitySettingBinding.inflate(layoutInflater)
    }

    override fun initVariable(savedInstanceState: Bundle?) {
        mRootLayout = getViewBinding().layoutRoot
        mToolbar = getViewBinding().toolbar

        if (!registered) {
            subscription = RxBus.getDefault().toObserverable(EventContainer::class.java)
                .subscribe { eventContainer ->
                    if (eventContainer.type == EventContainer.TYPE_SETTING) {
                        val event = eventContainer.event as SettingEvent
                        showSnackBar(mRootLayout, event.msg)
                    }
                }

            registered = true
        }

        initToolbar()
    }

    override fun doAction() {
        fragmentManager.beginTransaction().replace(R.id.content, SettingFragment()).commit()
    }

    private fun initToolbar() {
        mToolbar.setTitle(R.string.setting)
        setSupportActionBar(mToolbar)

        mToolbar.setNavigationOnClickListener(View.OnClickListener { finish() })
    }

    override fun onDestroy() {
        super.onDestroy()
        subscription?.unsubscribe()
    }
}
